ci: create ci workflow for multichain flow

cortisiko opened the pull request:
<!--
Please submit this PR as a draft initially.
Do not mark it as "Ready for review" until the template has been completely filled out, and PR status checks have passed at least once.
-->

## **Description**

This PR addresses the new flows introduced with the updated network chain permissions functionality. The goal is to ensure comprehensive automated test coverage for these changes, reducing the risk of bugs found later in the development process. 

Key things to note here:
breakdown of the CI workflows:

 - Smoke Pipeline: This pipeline runs all E2E tests with an enabled multichain flag. This ensures that engineers actively validate the new MC flows as part of their workflow. By default, the smoke pipeline runs on pull requests, allowing us to catch any issues with the MC flow before changes are merged into the main branch. We plan to make the smoke E2E pipeline a required step before merging PRs soon. 

 - Regression Pipeline: This pipeline runs E2E tests with the multichain flag disabled. It ensures that existing functionality remains stable without the MC changes. The regression pipeline is scheduled to run every four hours against the main branch, providing frequent and consistent validation of the baseline functionality.


## **Related issues**

Fixes:

## **Manual testing steps**

1. Go to this page...
2.
3.

## **Screenshots/Recordings**

<!-- If applicable, add screenshots and/or recordings to visualize the before and after of your change. -->

### **Before**

<!-- [screenshots/recordings] -->

### **After**

<!-- [screenshots/recordings] -->

## **Pre-merge author checklist**

- [ ] I’ve followed [MetaMask Contributor Docs](https://github.com/MetaMask/contributor-docs) and [MetaMask Mobile Coding Standards](https://github.com/MetaMask/metamask-mobile/blob/main/.github/guidelines/CODING_GUIDELINES.md).
- [ ] I've completed the PR template to the best of my ability
- [ ] I’ve included tests if applicable
- [ ] I’ve documented my code using [JSDoc](https://jsdoc.app/) format if applicable
- [ ] I’ve applied the right labels on the PR (see [labeling guidelines](https://github.com/MetaMask/metamask-mobile/blob/main/.github/guidelines/LABELING_GUIDELINES.md)). Not required for external contributors.

## **Pre-merge reviewer checklist**

- [ ] I've manually tested the PR (e.g. pull and build branch, run the app, test code being changed).
- [ ] I confirm that this PR addresses all acceptance criteria described in the ticket it closes and includes the necessary testing evidence such as recordings and or screenshots.

NidhiKJha committed:
updated failing test for multichain on feature flag removal

cortisiko committed:
remove multichain variables in code

cortisiko committed:
enable MC permission system

cortisiko committed:
set correct env variable path

cortisiko committed:
run 1 e2e test against the new workflow for MC

cortisiko committed:
trigger new workflow

cortisiko committed:
fix path

cortisiko committed:
these tests should fail because multichain is off and we are testing with the new tests

cortisiko committed:
enable it again

cortisiko committed:
comment out env variables

cortisiko committed:
Revert "comment out env variables"

This reverts commit c9cecd1c0154cecc38c097b9a3c7763daab3983c.

EtherWizard33 commented on app/util/networks/index.js:
```diff
@@ -584,13 +584,12 @@ export const deprecatedGetNetworkId = async () => {
 };
 
-export const isMultichainVersion1Enabled =
-  process.env.MM_MULTICHAIN_V1_ENABLED === 'true';
```
Hi @cortisiko  
this files need to stay like it is currently main, it is the intended code

and the code works fine in bitrise.yml file when I build a QA build, like in this build where I set the values in bitrise.yml

https://github.com/MetaMask/metamask-mobile/compare/main...build-qa-of-main-nov-25

cortisiko committed:
create qa builds with flag turned off

cortisiko committed:
kick off builds

cortisiko committed:
turn on flag now

NidhiKJha committed:
added e2e tests for chain permissions

cortisiko committed:
run e2e

cortisiko committed:
kick off new multichain tests

cortisiko committed:
update test to have SmokeCore tag for now

cortisiko committed:
use env variables correctly

cortisiko committed:
fix e2e and run on bitrise with variant on

cortisiko committed:
make turning on the mc flag for e2e better

cortisiko committed:
test multichain flag off

cortisiko committed:
Merge branch 'test-multichain-e2e-ci-workflow' into test-multichain-e2e-ci-workflow-tests

cortisiko committed:
Revert "test multichain flag off"

This reverts commit 18cc015634db508795eae83c56a84a579d121c7b.

EtherWizard33 committed:
test: update snapshots to match the new added test ids

EtherWizard33 committed:
Merge branch 'test-multichain-e2e-ci-workflow' into test-multichain-e2e-ci-workflow-eric-2

EtherWizard33 committed:
feat: add e2e test to revoke one chain permission (wip)

NidhiKJha committed:
feat: Test multichain e2e ci workflow tests (#12441)

<!--
Please submit this PR as a draft initially.
Do not mark it as "Ready for review" until the template has been
completely filled out, and PR status checks have passed at least once.
-->

## **Description**

<!--
Write a short description of the changes included in this pull request,
also include relevant motivation and context. Have in mind the following
questions:
1. What is the reason for the change?
2. What is the improvement/solution?
-->

## **Related issues**

Fixes:

## **Manual testing steps**

1. Go to this page...
2.
3.

## **Screenshots/Recordings**

<!-- If applicable, add screenshots and/or recordings to visualize the
before and after of your change. -->

### **Before**

<!-- [screenshots/recordings] -->

### **After**

<!-- [screenshots/recordings] -->

## **Pre-merge author checklist**

- [ ] I’ve followed [MetaMask Contributor
Docs](https://github.com/MetaMask/contributor-docs) and [MetaMask Mobile
Coding
Standards](https://github.com/MetaMask/metamask-mobile/blob/main/.github/guidelines/CODING_GUIDELINES.md).
- [ ] I've completed the PR template to the best of my ability
- [ ] I’ve included tests if applicable
- [ ] I’ve documented my code using [JSDoc](https://jsdoc.app/) format
if applicable
- [ ] I’ve applied the right labels on the PR (see [labeling
guidelines](https://github.com/MetaMask/metamask-mobile/blob/main/.github/guidelines/LABELING_GUIDELINES.md)).
Not required for external contributors.

## **Pre-merge reviewer checklist**

- [ ] I've manually tested the PR (e.g. pull and build branch, run the
app, test code being changed).
- [ ] I confirm that this PR addresses all acceptance criteria described
in the ticket it closes and includes the necessary testing evidence such
as recordings and or screenshots.

---------

Co-authored-by: Curtis <Curtis.David7@gmail.com>
Co-authored-by: EtherWizard33 <eric.lamontagne@consensys.net>

cortisiko committed:
Merge branch 'main' into test-multichain-e2e-ci-workflow

EtherWizard33 committed:
test: add e2e test that revokes a network permission

EtherWizard33 committed:
Merge branch 'test-multichain-e2e-ci-workflow' into test-multichain-e2e-ci-workflow-eric-2

EtherWizard33 committed:
style: lint

EtherWizard33 committed:
test: update snapshot after updated test ID

EtherWizard33 committed:
chore: bitrise toggle chain permissions feature flag on in bitrise config (#12473)

<!--
Please submit this PR as a draft initially.
Do not mark it as "Ready for review" until the template has been
completely filled out, and PR status checks have passed at least once.
-->

## **Description**
After having obtained the go ahead, toggle the chain permissions feature
flag on.
<!--
Write a short description of the changes included in this pull request,
also include relevant motivation and context. Have in mind the following
questions:
1. What is the reason for the change?
2. What is the improvement/solution?
-->

## **Related issues**

Fixes:

## **Manual testing steps**

## **Screenshots/Recordings**

<!-- If applicable, add screenshots and/or recordings to visualize the
before and after of your change. -->

### **Before**

<!-- [screenshots/recordings] -->

### **After**

<!-- [screenshots/recordings] -->

## **Pre-merge author checklist**

- [ ] I’ve followed [MetaMask Contributor
Docs](https://github.com/MetaMask/contributor-docs) and [MetaMask Mobile
Coding
Standards](https://github.com/MetaMask/metamask-mobile/blob/main/.github/guidelines/CODING_GUIDELINES.md).
- [ ] I've completed the PR template to the best of my ability
- [ ] I’ve included tests if applicable
- [ ] I’ve documented my code using [JSDoc](https://jsdoc.app/) format
if applicable
- [ ] I’ve applied the right labels on the PR (see [labeling
guidelines](https://github.com/MetaMask/metamask-mobile/blob/main/.github/guidelines/LABELING_GUIDELINES.md)).
Not required for external contributors.

## **Pre-merge reviewer checklist**

- [ ] I've manually tested the PR (e.g. pull and build branch, run the
app, test code being changed).
- [ ] I confirm that this PR addresses all acceptance criteria described
in the ticket it closes and includes the necessary testing evidence such
as recordings and or screenshots.

cortisiko committed:
Enable mc flag off regression (#12475)

<!--
Please submit this PR as a draft initially.
Do not mark it as "Ready for review" until the template has been
completely filled out, and PR status checks have passed at least once.
-->

## **Description**

The purpose of this PR is to run the multichain variant flag off on
regression builds.
## **Related issues**

Fixes:

## **Manual testing steps**

1. Go to this page...
2.
3.

## **Screenshots/Recordings**

<!-- If applicable, add screenshots and/or recordings to visualize the
before and after of your change. -->

### **Before**

<!-- [screenshots/recordings] -->

### **After**

<!-- [screenshots/recordings] -->

## **Pre-merge author checklist**

- [ ] I’ve followed [MetaMask Contributor
Docs](https://github.com/MetaMask/contributor-docs) and [MetaMask Mobile
Coding
Standards](https://github.com/MetaMask/metamask-mobile/blob/main/.github/guidelines/CODING_GUIDELINES.md).
- [ ] I've completed the PR template to the best of my ability
- [ ] I’ve included tests if applicable
- [ ] I’ve documented my code using [JSDoc](https://jsdoc.app/) format
if applicable
- [ ] I’ve applied the right labels on the PR (see [labeling
guidelines](https://github.com/MetaMask/metamask-mobile/blob/main/.github/guidelines/LABELING_GUIDELINES.md)).
Not required for external contributors.

## **Pre-merge reviewer checklist**

- [ ] I've manually tested the PR (e.g. pull and build branch, run the
app, test code being changed).
- [ ] I confirm that this PR addresses all acceptance criteria described
in the ticket it closes and includes the necessary testing evidence such
as recordings and or screenshots.

EtherWizard33 committed:
test: update snapshots to take into consideration added test ids

cortisiko committed:
test: create new mc test (#12477)

<!--
Please submit this PR as a draft initially.
Do not mark it as "Ready for review" until the template has been
completely filled out, and PR status checks have passed at least once.
-->

## **Description**

Scenario:
Connect to dapp with custom network
Go to wallet 
Remove custom network
Return to dapp
Confirm the network permission sheet appears

## **Related issues**

Fixes:

## **Manual testing steps**

1. Go to this page...
2.
3.

## **Screenshots/Recordings**

<!-- If applicable, add screenshots and/or recordings to visualize the
before and after of your change. -->

### **Before**

<!-- [screenshots/recordings] -->

### **After**

<!-- [screenshots/recordings] -->

## **Pre-merge author checklist**

- [ ] I’ve followed [MetaMask Contributor
Docs](https://github.com/MetaMask/contributor-docs) and [MetaMask Mobile
Coding
Standards](https://github.com/MetaMask/metamask-mobile/blob/main/.github/guidelines/CODING_GUIDELINES.md).
- [ ] I've completed the PR template to the best of my ability
- [ ] I’ve included tests if applicable
- [ ] I’ve documented my code using [JSDoc](https://jsdoc.app/) format
if applicable
- [ ] I’ve applied the right labels on the PR (see [labeling
guidelines](https://github.com/MetaMask/metamask-mobile/blob/main/.github/guidelines/LABELING_GUIDELINES.md)).
Not required for external contributors.

## **Pre-merge reviewer checklist**

- [ ] I've manually tested the PR (e.g. pull and build branch, run the
app, test code being changed).
- [ ] I confirm that this PR addresses all acceptance criteria described
in the ticket it closes and includes the necessary testing evidence such
as recordings and or screenshots.

cortisiko committed:
Merge branch 'main' into test-multichain-e2e-ci-workflow

tommasini reviewed (dismissed):
Awesome!

cortisiko committed:
build regression, set mc flag to false

cortisiko committed:
move portfolio test to quarantine for now. Will follow up with a PR ASAP to fix
